spec/spec_helper_acceptance.rb

```diff
@@ -13,6 +13,14 @@
 install_module_on(hosts)
 install_module_dependencies_on(hosts)
 
+# Bolt helper task
+def task_run(executor, params)
+  module_path = RSpec.configuration.module_path
+  config = { 'modulepath' => module_path }
+  inventory = hosts_to_inventory.merge('features' => ['puppet-agent'])
```

Comment:
This line needs removed, it looks like accidental debug leftover from moving stuff from puppet_agent. Instead of this line you can just update the run_task call below like so: 
  run_task(executor, 'default', params, config: config, inventory: hosts_to_inventory)

Reply:
Setting the `puppet-agent` feature in inventory ensures that the ruby implementation of the task is run. That was intentional in the init_spec.rb test file. 

In general the task acceptance tests were set up to be a "starting point" for development. As the tasks are developed and more robust tests are written I think we will want to configure `modulepath`, `config`, and `inventory` within the individual test files. For example a `windows_spec.rb` set of tests *should* be added to test the powershell implementation. 

If anything I think a refactor that simply moved the `modulepath`, `config`, and `inventory` functions to `spec_helper_acceptance/rb`  could be a viable option. I am having a hard time seeing any value in a generic `run` function in the spec helper. As implemented in this PR there is no control over the target, config, or inventory. Also the parameter name `executor` is especially confusing given it has a very specific meaning in bolt (something like `task_name` would be more accurate the way it is implemented here).

Reply:
https://tickets.puppetlabs.com/browse/BOLT-1103 is in the Bolt team queue. Part of that work will be to unify the output for the different task implementations. That work should include some new test development which might help inform decisions on refactoring the test setup.

Reply:
Hi @donoghuc . We plan to convert these modules to use Litmus in the future. It has a built in function for running tasks called `task_run`. When it comes to converting the modules we can simply cut things out such as this helper it will (hopefully!) work out of the box with few code changes. The inventory and configuration will also become part of the inventory file used by Litmus so these will no longer be needed in the tests. The Litmus repo is open now so I'd recommend checking it out if you haven't already!

Reply:
I'm not sure I see how the `run_task` implementation will work here https://github.com/puppetlabs/puppet_litmus/blob/f858434e90e3c52138e1482f9a186024e8863a57/lib/puppet_litmus.rb#L148-L160

Specifically:
1. How will we modify inventory for example add a feature to a target or goup, with a tempfile? 
2. How will you specify a target? Setting an environment variable? 
3. How can you modify config?

Also, using methods from `Beaker::TaskHelper::Inventory` and `BoltSpec::Run` in `spec_helper_acceptance.rb` https://github.com/puppetlabs/puppetlabs-service/blob/19e76e66efcb2d5d04f021a8f5c05015c5a873fa/spec/spec_helper_acceptance.rb#L16-L22 without including those modules explicitly forces those to be included in any of the specs that use `task_run` for example
https://github.com/puppetlabs/puppetlabs-service/blob/19e76e66efcb2d5d04f021a8f5c05015c5a873fa/spec/acceptance/init_spec.rb#L5-L6 and https://github.com/puppetlabs/puppetlabs-service/blob/19e76e66efcb2d5d04f021a8f5c05015c5a873fa/spec/acceptance/linux_spec.rb#L5-L6
 I think it is OK to just include those in `spec_helper_acceptance.rb`. 

Finally, I think that some of the commits in this PR could have been squashed or omitted and could have used better messages and perhaps the (MODULES) ticket in the commit message title. 

I'm working on a refactor of this module and adding tests for the powershell implementation (https://github.com/puppetlabs/puppetlabs-service/pull/83). I am OK using your implementation of `task_run` if it will help future proof for Litmus. I am rebasing on what you have committed to master and will make updates as needed. I will request review once it becomes ready for merge. Thanks @eimlav @lionce